AtReconstruction/AtPatternModification/AtPatternModification.h

```diff
@@ -0,0 +1,38 @@
+#ifndef AtPATTERNMODIFICATION_H
+#define AtPATTERNMODIFICATION_H
+
+#include "AtEvent.h"
+#include "AtPatternEvent.h"
+#include "AtRawEvent.h"
+
+class AtPatternModification : public TObject {
+protected:
+   // Pointer to the AtPatternEvent to be modified.
+   AtPatternEvent *fPatternEvent{nullptr};
+
+   // Pointers to AtRawEvent and AtEvent that may be used or not in the modification.
+   AtRawEvent *fRawEvent{nullptr};
+   AtEvent *fEvent{nullptr};
+
+public:
+   AtPatternModification() = default;
+   ~AtPatternModification() = default;
+
+   virtual void ModifyPatternEvent() = 0;
+   virtual void Init() = 0;
+
+   // Mandatory to set.
+   void SetPatternEvent(AtPatternEvent *patternEvent) { fPatternEvent = patternEvent; }
+
+   // Optional to set.
+   void SetRawEvent(AtRawEvent *rawEvent) { fRawEvent = rawEvent; }
+   void SetEvent(AtEvent *event) { fEvent = event; }
+
+   // Reset pointers.
+   void Reset();
```

Comment:
I think this interface is overly cumbersome. It requires the user to set and reset various pointers, when that seems like it is unnecessary. I think this entire interface should be:

```cpp
#ifndef AtPATTERNMODIFICATION_H
#define AtPATTERNMODIFICATION_H

class AtEvent.h;
class AtPatternEvent.h;
class AtRawEvent.h;

class AtPatternModification {
virtual void Init() = 0;
/**
* Modifies patternEvent.
*/
virtual void ModifyPatternEvent(AtPatternEvent *patternEvent, const AtRawEvent *raw=nullptr, const AtEvent* event = nullptr) = 0;

};